From: Borislav Petkov <bp@suse.de>
Date: Fri May 25 19:24:16 CEST 2018
Subject: kabi: Fix kvm kABI
Patch-mainline: no, SUSE-specific
References: bsc#1087082 CVE-2018-3639

... due to

  bc226f07dcd3 ("KVM: SVM: Implement VIRT_SPEC_CTRL support for SSBD")

changing kvm_x86_ops' member.

Signed-off-by: Borislav Petkov <bp@suse.de>

---
 arch/x86/include/asm/kvm_host.h |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -813,7 +813,7 @@ struct kvm_x86_ops {
 	int (*hardware_setup)(void);               /* __init */
 	void (*hardware_unsetup)(void);            /* __exit */
 	bool (*cpu_has_accelerated_tpr)(void);
-	bool (*has_emulated_msr)(int index);
+	bool (*cpu_has_high_real_mode_segbase)(void);
 	void (*cpuid_update)(struct kvm_vcpu *vcpu);
 
 	int (*vm_init)(struct kvm *kvm);
@@ -975,6 +975,7 @@ struct kvm_x86_ops {
 #ifndef __GENKSYMS__
 	void (*set_pkru)(struct kvm_vcpu *vcpu, u32 pkru);
 	bool (*get_enable_apicv_new)(struct kvm_vcpu *vcpu);
+	bool (*has_emulated_msr)(int index);
 #endif
 };
 
